gapis/memory/memory_pb/memory.proto

```diff
@@ -19,32 +19,22 @@ package memory_pb;
 
 // Observation is the structure for a single memory observation.
 message Observation {
+    // The pool identifier.
+    uint64 pool = 1;
     // Base is the starting address for the observation range.
-    uint64 base = 1;
+    uint64 base = 2;
     // Size is the byte count of the observation range.
-    uint64 size = 2;
-    // Id is the resource id of the bytes observed (used in memory).
-    bytes id = 3;
+    uint64 size = 3;
     // ResIndex is the index of the resource (used on disk).
     sint64 resIndex = 4;
 }
 
-// Pointer is the type representing a memory pointer.
-message Pointer {
-    // The memory address.
-    uint64 Address = 1;
-    // The memory pool.
-    uint32 Pool = 2;
-}
-
 // Slice is the common data between all slice types.
 message Slice {
-    // Original pointer this slice derives from.
-    uint64 Root = 1;
+    // The pool identifier.
```

Comment:
Don't remove `Root`. As it is currently required by GAPIS, it is required to be encoded by GAPII for MEC. Without this information you cannot always reconstruct the replay memory buffers correctly.

Reply:
I know what is the use of Root in gapis, but gapii does not use it, and it does not make any sense to me on-disk, or inside the state block.  I don't really see why it would be needed for MEC, and Andrew say he does not see any need for it in MEC either.

My thinking is that I don't want to add extra stuff in the MEC state block, if we don't use it now.  And it is easy to add the field back later if we find use for it.

Reply:
If you persisted a state block that has slices to the application pool for vertex attribute pointers, you will need it. That's why it exists.

Reply:
I think we need to persist those.

So, if we keep it in, is the value of Root going to be == Base for all slices for now?
Or do we omit the Root (i.e. 0) if it is not used to save space?

Reply:
I'd say Root == Base. I don't think there's any point optimizing something like that for the MEC state block.

Reply:
Fair point. ok